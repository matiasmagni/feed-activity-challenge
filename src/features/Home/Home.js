import React from 'react';
import PostFeeds from '../PostFeeds/PostFeeds';

const Home = () => {
    return (
        <>
            <PostFeeds />

            <footer className="container-fluid pt-5">
                <div className="container">
                    <h2 className="logo text-center">Shahala</h2>
                    <nav className="nav nav-fill mx-auto mt-5">
                        <li className="nav-item">
                            <a className="nav-link" href="https://facebook.com/fh5co" target="_blank" rel="noreferrer"><i
                                className="fab fa-facebook-f"></i></a>
                        </li>
                        <li className="nav-item">
                            <a className="nav-link" href="https://facebook.com/fh5co" target="_blank" rel="noreferrer"><i
                                className="fab fa-twitter"></i></a>
                        </li>
                        <li className="nav-item">
                            <a className="nav-link" href="/#"><i className="fab fa-instagram"></i></a>
                        </li>
                        <li className="nav-item">
                            <a className="nav-link" href="/#"><i className="fab fa-google-plus-g"></i></a>
                        </li>
                        <li className="nav-item">
                            <a className="nav-link" href="/#"><i className="fas fa-rss"></i></a>
                        </li>
                    </nav>
                </div>
                <div className="copyright mt-4">
                    <p className="text-center">&copy; 2019 <a href="/#" className="text-white">Shahala</a>. All Rights Reserved. Design
                by <a href="https://freehtml5.co/shahala" target="_blank" rel="noreferrer" className="text-white">FreeHTML5.co</a>.</p>
                </div>
            </footer>
        </>
    );
}

export default Home;
