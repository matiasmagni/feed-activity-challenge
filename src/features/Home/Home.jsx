import React from 'react';
import PostFeeds from '../PostFeeds/PostFeeds';

const Home = () => (
  <>
    <PostFeeds />

    <footer className="container-fluid pt-5">
      <div className="container">
        <h2 className="logo text-center">Shahala</h2>
        <nav className="nav nav-fill mx-auto mt-5">
          <li className="nav-item">
            <a className="nav-link" href="https://facebook.com/fh5co" target="_blank" rel="noreferrer">
              <i
                className="fab fa-facebook-f"
              />
            </a>
          </li>
          <li className="nav-item">
            <a className="nav-link" href="https://facebook.com/fh5co" target="_blank" rel="noreferrer">
              <i
                className="fab fa-twitter"
              />
            </a>
          </li>
          <li className="nav-item">
            <a className="nav-link" href="/#"><i className="fab fa-instagram" /></a>
          </li>
          <li className="nav-item">
            <a className="nav-link" href="/#"><i className="fab fa-google-plus-g" /></a>
          </li>
          <li className="nav-item">
            <a className="nav-link" href="/#"><i className="fas fa-rss" /></a>
          </li>
        </nav>
      </div>
      <div className="copyright mt-4">
        <p className="text-center">
          &copy; 2019
          <a href="/#" className="text-white">Shahala</a>
          . All Rights Reserved. Design
          by
          <a href="https://freehtml5.co/shahala" target="_blank" rel="noreferrer" className="text-white">FreeHTML5.co</a>
          .
        </p>
      </div>
    </footer>
  </>
);

export default Home;
