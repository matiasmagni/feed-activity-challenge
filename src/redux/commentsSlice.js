import { createSlice } from '@reduxjs/toolkit';

export const commentsSlice = createSlice({
    name: 'comments',
    initialState: {
        comments: [],
    },
    reducers: {
        setAllComments: (state, action) => {
            state.comments.push(action.payload.comments);
        },
        addComment: (state, action) => {
            state.comments.push(action.payload);
        },
    }
});

export const { setAllComments, addComment } = commentsSlice.actions;
export const selectComments = state => state.comments.comments;

export default commentsSlice.reducer;
