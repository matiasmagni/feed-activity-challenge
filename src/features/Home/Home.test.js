import React from 'react';
import { cleanup } from '@testing-library/react';
import { act as domAct } from "react-dom/test-utils";
import { Provider } from 'react-redux';
import { act as testAct, create } from "react-test-renderer";
import { store } from '../../redux/store';
import Home from './Home';

afterEach(cleanup);

describe('Home component tests.', () => {
  it('Home component renders correctly.', () => {
    let tree;
  
    domAct(() => {
      testAct(() => {
        tree = create(<Provider store={store}><Home /></Provider>);
      });
  
      expect(tree).toMatchSnapshot();
    });
  });
});
