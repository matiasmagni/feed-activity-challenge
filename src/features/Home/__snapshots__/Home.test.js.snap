// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component tests. Home component renders correctly. 1`] = `
Array [
  <div
    className="container-fluid video-player"
  >
    <div
      className="container"
    >
      <div
        className="row vr-gallery"
      />
    </div>
  </div>,
  <footer
    className="container-fluid pt-5"
  >
    <div
      className="container"
    >
      <h2
        className="logo text-center"
      >
        Shahala
      </h2>
      <nav
        className="nav nav-fill mx-auto mt-5"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="https://facebook.com/fh5co"
            rel="noreferrer"
            target="_blank"
          >
            <i
              className="fab fa-facebook-f"
            />
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="https://facebook.com/fh5co"
            rel="noreferrer"
            target="_blank"
          >
            <i
              className="fab fa-twitter"
            />
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/#"
          >
            <i
              className="fab fa-instagram"
            />
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/#"
          >
            <i
              className="fab fa-google-plus-g"
            />
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/#"
          >
            <i
              className="fas fa-rss"
            />
          </a>
        </li>
      </nav>
    </div>
    <div
      className="copyright mt-4"
    >
      <p
        className="text-center"
      >
        Â© 2019 
        <a
          className="text-white"
          href="/#"
        >
          Shahala
        </a>
        . All Rights Reserved. Design by 
        <a
          className="text-white"
          href="https://freehtml5.co/shahala"
          rel="noreferrer"
          target="_blank"
        >
          FreeHTML5.co
        </a>
        .
      </p>
    </div>
  </footer>,
]
`;
